doc/glossary.rst

```diff
@@ -0,0 +1,973 @@
+        estimator-as-parameter and its parameter.  Thus
+        ``BaggingClassifier(base_estimator=DecisionTreeClassifier(max_depth=3))``
+        has a deep parameter ``base_estimator__max_depth`` with value ``3``.
+
+        The list of parameters and their current values can be retrieved from
+        an :term:`estimator instance` using its :term:`get_params` method.
+
+        Between construction and fitting, parameters may be modified using
+        :term:`set_params`.  To enable this, parameters are not ordinarily
+        validated or altered when the estimator is constructed, or when each
+        parameter is set. Parameter validation is performed when :term:`fit` is
+        called.
+
+        Common parameters are listed :ref:`below <glossary_attributes>`.
+
+    pairwise metric
+        TODO
+
+        See precomputed.
+
+    precomputed
+        TODO
+
+    rectangular
+        Data that can be represented as a matrix with :term:`samples` on the
+        first axis and a fixed, finite set of :term:`features` on the second
+        is called rectangular.
+
+        This term excludes samples with non-vectorial structure, such as text,
+        an image of arbitrary size, a time series of arbitrary length, a set of
+        vectors, etc. The purpose of a :term:`vectorizer` is to produce
+        rectangular forms of such data.
+
+    sample
+    samples
+        We usually use this term as a noun to indicate a single feature vector.
+        Elsewhere a sample is called an instance, data point, or observation.
+        ``n_samples`` indicates the number of samples in a dataset.
+
+    sample property
+    sample properties
+        TODO
+
+    scikit-learn-contrib
+        TODO
+
+    scorer
+        TODO
+        See :ref:`scoring_parameter`.
+        See also :term:`evaluation metric`.
+
+    sparse matrix
+        TODO
+
+    target
+    targets
+        TODO
+
+        Dependent variable or outcome variable.
+
+    unlabeled
+    unlabeled data
+        TODO
+
+Class APIs and Estimator Types
+==============================
+
+.. glossary::
+
+    classifier
+        TODO
+        Mention that within scikit-learn, all support multi-class
+        classification, defaulting to OvR.
+        Mention :func:`~base.is_classifier`.
+
+    clusterer
+        TODO
+
+    cross validation splitter
+    CV splitter
+        TODO
+
+    estimator
+        TODO
+
+        The core functionality of some estimators may also be available as a
+        :term:`function`.
+
+    feature extractor
+        A :term:`tranformer` which takes input where each sample is not
+        represented as an :term:`array-like` object of fixed length, and
+        produces an `array-like` object of :term:`features` for each sample
+        (and thus a 2-dimensional array-like for a set of samples).  In other
+        words, it (lossily) maps a non-rectangular data representation into
+        :term:`rectangular` data.
+
+    meta-estimator
```

Comment:
Hm I guess when I started reading I wasn't clear on the separation into the different lists. Maybe ctrl+f is enough for users to find what they are looking for, but I'm not sure ...

Reply:
Yeah...